core/src/main/java/nl/nn/adapterframework/pipes/FixedResultPipe.java

```diff
@@ -42,15 +41,35 @@
 import nl.nn.adapterframework.util.TransformerPool;
 
 /**
- * Provides an example of a pipe. It may return the contents of a file
- * (in the classpath) when <code>filename</code> or <code>filenameSessionKey</code> is specified, otherwise the
- * input of <code>returnString</code> is returned.
+ * Produces a fixed result that does not depend on the input message. It may return the contents of a file
+ * when <code>filename</code> or <code>filenameSessionKey</code> is specified. Otherwise the
+ * value of attribute <code>returnString</code> is returned.
+ * <br/><br/>
+ * Using parameters and the attributes of this pipe, it is possible to substitute values. This pipe
+ * performs the following steps:
+ * <ol>
+ * <li>When a configuration is loaded, all expressions <code>${...}</code> within <code>Configuration.xml</code> and its referenced entities are substituted.
+ * This substitution is done according to system properties and application properties (DeploymentSpecifics.properties and others). See the Frank!Manual for details.
+ * This substitution is done before this pipe is configured or executed.
+ * <li>During execution, this pipe first obtains a string based on attributes <code>returnString</code>, <code>filename</code> or <code>filenameSessionKey</code>.
+ * <li>The resulting string is transformed according to attributes <code>replaceFrom</code> and <code>replaceTo</code> if set.
+ * Please note that the plain value of attribute <code>replaceFrom</code> is matched, no <code>${...}</code> here.
+ * <li>The resulting string is substituted based on the parameters of this pipe. This step depends on attribute <code>replaceFixedParams</code>.
+ * Assume that there is a parameter with name <code>xyz</code>. If <code>replaceFixedParams</code> is <code>false</code>, then
+ * each occurrence of <code>${xyz}</code> is replaced by the parameter's value. Otherwise, the text <code>xyz</code>
+ * is substituted. See {@link nl.nn.adapterframework.parameters.Parameter} to see how parameter values are determined.
+ * <li>If attribute <code>substituteVars</code> is <code>true</code>, then expressions <code>${...}</code> are substituted using
+ * system properties, pipelinesession variables and application properties. Please note that
+ * no <code>${...}</code> patterns are left if the initial string came from attribute <code>returnString</code>, because
+ * any <code>${...}</code> pattern in attribute <code>returnString</code> is substituted when the configuration is loaded.
```

Comment:
Het is handig hier expliciet te melden dat het gaat om `${...}` waardes in de file die gebruikt wordt, als je dit in attribuut returnString zet wordt dit tijdens het 'parsen' (lezen) van de configuratie gedaan.